pkg/server/containerssh.go

```diff
@@ -61,16 +69,23 @@ func (s *Server) OnPubKey(c *gin.Context) {
 		c.JSON(500, err)
 		return
 	}
+
+	owner, _, err := sshname.GetInfo(req.Username)
+	if err != nil {
+		c.JSON(500, err)
+		return
+	}
+
 	for _, k := range s.authInfo {
-		logrus.Info(k.PublicKey, k.IdentityToken)
 		key, _, _, _, err := ssh.ParseAuthorizedKey([]byte(req.PublicKey.PublicKey))
 		if err != nil {
 			logrus.WithError(err).Error("failed to parse key")
 			c.JSON(500, err)
 			return
 		}
 		// https://github.com/golang/go/issues/21704#issuecomment-342760478
-		if bytes.Equal(key.Marshal(), k.PublicKey.Marshal()) {
+		if owner == k.IdentityToken &&
+			bytes.Equal(key.Marshal(), k.PublicKey.Marshal()) {
```

Comment:
I think it should use https://pkg.go.dev/crypto/subtle#ConstantTimeCompare

Reply:
SGTM